### PR TITLE
[CustomOp] Support install as Package and Add load interface

### PR DESCRIPTION
**Pull request by Aurelius84**
### PR types

New features
### PR changes

APIs
### Architecture

Architecture

#### 1. Move into `Paddle.utils`
All cpp extension script files have been moved into `paddle.utils.cpp_extension`

#### 2. Support to install as package
Support users to install custom op as isolate python package, and call it **just only with two line**:

```python
import custom_op_shared

out = custom_op_shared.relu2(x)
```

#### 3. Support `load` to jit compile .cc/.cu
Support JIT way  to hidden inner complilation process.
```python
from paddle.utils.cpp_extension import load

relu2  = load('relu2_shared', ['relu_op.cc', 'relu_op.cu'])

out = relu2(x)
```

### CODE REVIEW

File: python/paddle/utils/cpp_extension/cpp_extension.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+    assert isinstance(cmdclass, dict)
+    # if not specific cmdclass in setup, add it automaticaly.
+    if 'build_ext' not in cmdclass:
+        cmdclass['build_ext'] = BuildExtension.with_options(
+            no_python_abi_suffix=True)
+        attr['cmdclass'] = cmdclass
+    # elif not isinstance(cmdclass['build_ext'], BuildExtension):
+    #     raise ValueError(
+    #         "Require paddle.utils.cpp_extension.BuildExtension in setup(cmdclass={'build_ext: ...'}), but received {}".
+    #         format(type(cmdclass['build_ext'])))
```
Comment (Aurelius84) [inline python/paddle/utils/cpp_extension/cpp_extension.py:49]: I will fix this in next PR.

**Review by XieYunshen — APPROVED** [review]

**Review by luotao1 — APPROVED** [review]

**Review by chenwhql — APPROVED** [review]
LGTM

**Review by zhwesky2010 — APPROVED** [review]

**Review by TCChenlong — APPROVED** [review]
LGTM

**Review by Heeenrrry — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]

**Pull request merged**